Sources/Signal.swift

```diff
@@ -118,6 +132,22 @@ public final class Signal<Value, Error: Swift.Error> {
 					self.stateLock.unlock()
 				}
 
+				while true {
```

Comment:
This doesn't seem like the best way to implement this, but I guess you can't do `while !self.buffer.isEmpty` cause you need to lock?

Reply:
On top of locking, the signal needs to be alive before us proceeding.